src/Microsoft.AspNet.Mvc.Razor/DefaultViewLocationCache.cs

```diff
@@ -49,7 +50,9 @@ internal static string GenerateKey(ViewLocationExpanderContext context)
         {
             var keyBuilder = new StringBuilder();
             var routeValues = context.ActionContext.RouteData.Values;
-            var controller = routeValues.GetValueOrDefault<string>(RazorViewEngine.ControllerKey);
+            var controller = RouteUtility.GetNormalizedRouteValue(
+                RazorViewEngine.ControllerKey,
+                context.ActionContext);
 
             // format is "{viewName}:{isPartial}:{controllerName}:{areaName}:"
             keyBuilder.Append(context.ViewName)
```

Comment:
I think the view name here is a user-supplied one? if yes, there are chances the case could be different depending on different areas of code executing this view (even though the chances are remote i think)...one alternative is to make the dictionary key to be `OrdinalIgnoreCase` and you can avoid doing all the other changes in this class...